File: Basic Coding Problems/02 If - Elses/035 Write a program to find all roots of a quadratic equation/C++/quadratic_equation.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {
    float a, b, c;
    float discriminant, root1, root2, real_part, imaginary_part;

    cout << "Enter the value of a, b, c: ";
    cin >> a >> b >> c;

    discriminant = b * b - 4 * a * c;

    if (discriminant > 0) {
        root1 = (-b + sqrt(discriminant)) / (2 * a);
        root2 = (-b - sqrt(discriminant)) / (2 * a);
        cout << "The roots are: " << root1 << " and " << root2 << endl;
    } else if (discriminant == 0) {
        root1 = root2 = -b / (2 * a);
        cout << "The roots are: " << root1 << " and " << root2 << endl;
    } else {
        real_part = -b / (2 * a);
        imaginary_part = sqrt(-discriminant) / (2 * a);
        cout << "The roots are: " << real_part << " + " << imaginary_part << "i and " << real_part << " - " << imaginary_part << "i" << endl;
    }

    return 0;
}

File: Basic Coding Problems/02 If - Elses/035 Write a program to find all roots of a quadratic equation/Java/quadratic_equation.java
import java.util.Scanner;

public class quadratic_equation {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the value of a, b, c: ");
        float a = scanner.nextFloat();
        float b = scanner.nextFloat();
        float c = scanner.nextFloat();

        float discriminant = b * b - 4 * a * c;

        if (discriminant > 0) {
            float root1 = (float) ((-b + Math.sqrt(discriminant)) / (2 * a));
            float root2 = (float) ((-b - Math.sqrt(discriminant)) / (2 * a));
            System.out.println("The roots are: " + root1 + " and " + root2);
        } else if (discriminant == 0) {
            float root2;
            float root1 = root2 = -b / (2 * a);
            System.out.println("The roots are: " + root1 + " and " + root2);
        } else {
            float realPart = -b / (2 * a);
            float imaginaryPart = (float) (Math.sqrt(-discriminant) / (2 * a));
            System.out.println("The roots are: " + realPart + " + " + imaginaryPart + "i and " + realPart + " - " + imaginaryPart + "i");
        }
    }
}